### PR TITLE
tool: change coding style, indent with two spaces

### PR DESCRIPTION
**Pull request by simon-rock**
Signed-off-by: Simon Gao <simon29rock@gmail.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/tools/cephfs/MetaTool.cc [hunk 1]
```diff
@@ -36,28 +38,28 @@ WRITE_RAW_ENCODER(unsigned char)
...
-        delete i.second;
-    }
-    while(!sub_ops.empty()){
-        delete sub_ops.top();
-        sub_ops.pop();
-    }
+  for (const auto& i : inodes) {
+    delete i.second;
+  }
+  while(!sub_ops.empty()){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:44]: add space before `while`

File: src/tools/cephfs/MetaTool.cc [hunk 2]
```diff
@@ -68,263 +70,263 @@ int MetaTool::main(string& mode,
...
-                      << "\nstart manual mode!!"<< std::endl;
-            manual_mode = true;
-        }    
+  int r = 0;
+
+  std::string manual_meta_pool;
+  std::string manual_data_pool;
+  std::string manual_rank_num;
+  bool manual_mode = false;
+  if (minfo != ""){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:79]: space before `{`.

File: src/tools/cephfs/MetaTool.cc [hunk 3]
```diff
@@ -68,263 +70,263 @@ int MetaTool::main(string& mode,
...
+    vector<string> v;
+    string_split(minfo, v);
+    manual_meta_pool = v.size() >= 1?v[0]:"";
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:82]: spaces around `?` and `:`.

File: src/tools/cephfs/MetaTool.cc [hunk 4]
```diff
@@ -68,263 +70,263 @@ int MetaTool::main(string& mode,
...
+  }
+
+  // RADOS init
+  r = rados.init_with_context(g_ceph_context);
+  if (r < 0) {
+    cerr << "RADOS unavailable" << std::endl;
+    return r;
+  }
+
+  if(_debug)
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:106]: space after `if`.

File: src/tools/cephfs/MetaTool.cc [hunk 5]
```diff
@@ -68,263 +70,263 @@ int MetaTool::main(string& mode,
...
+    cout << "MetaTool: connecting to RADOS..." << std::endl;
+  r = rados.connect();
+  if (r < 0) {
+    cerr << "couldn't connect to cluster: " << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+  if (!manual_mode){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:114]: space before `{`.

File: src/tools/cephfs/MetaTool.cc [hunk 6]
```diff
@@ -68,263 +70,263 @@ int MetaTool::main(string& mode,
...
-        return r;
+    auto fs = fsmap->get_filesystem(role_selector.get_ns());
+    assert(fs != nullptr);
+
+    // prepare io for meta pool
+    int64_t const pool_id = fs->mds_map.get_metadata_pool();
+    features = fs->mds_map.get_up_features();
+    if (features == 0)
+      features = CEPH_FEATURES_SUPPORTED_DEFAULT;
+    else if (features != CEPH_FEATURES_SUPPORTED_DEFAULT){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:129]: space before `{`.

File: src/tools/cephfs/MetaTool.cc [hunk 7]
```diff
@@ -68,263 +70,263 @@ int MetaTool::main(string& mode,
...
+    }
+
+    if (_debug)
+      cout << "MetaTool: creating IoCtx.." << std::endl;
+    r = rados.ioctx_create(pool_name.c_str(), io_meta);
+    assert(r == 0);
+    output.dup(io_meta);
+
+    // prepare io for data pool
+    const std::vector<int64_t>  & data_pools_v = fs->mds_map.get_data_pools();
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:148]: remove spaces before `&`.

File: src/tools/cephfs/MetaTool.cc [hunk 8]
```diff
@@ -68,263 +70,263 @@ int MetaTool::main(string& mode,
...
+    for(auto i = data_pools_v.begin(); i != data_pools_v.end(); ++i){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:149]: use range-base loop.

File: src/tools/cephfs/MetaTool.cc [hunk 9]
```diff
@@ -68,263 +70,263 @@ int MetaTool::main(string& mode,
...
+    inodeno_t i_ino = conv2hexino(ino.c_str());
+    meta_op op(_debug, out);
+
+    meta_op::sub_op* nsop = new meta_op::sub_op(&op);
+    nsop->sub_op_t = meta_op::OP_SHOW;
+    nsop->sub_ino_t = meta_op::INO_DIR;
+    nsop->ino = i_ino;
+    op.push_op(nsop);
+    return op_process(op);
+  }else{
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:267]: spaces around `else`.

File: src/tools/cephfs/MetaTool.cc [hunk 10]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+    cout << "Error parsing create user response" << std::endl;
+    return -1;
+  }
+  try {
+    inode_meta.decode_json(&parser);
+  } catch (JSONDecoder::err& e) {
+    cout << "failed to decode JSON input: " << e.what() << std::endl;
+    return -1;
+  }
+  if (!op.confirm_chg() || op.is_debug()){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:349]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 11]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+    cout << "you will amend info of inode ==>: " << std::endl;
+    _show_meta(inode_meta, "");
+  }
+  if (!op.confirm_chg()){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:353]: space

File: src/tools/cephfs/MetaTool.cc [hunk 12]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+  bufferlist bl;
+  inode_meta.encode(bl, features);
+  map<string, bufferlist> to_set;
+  to_set[k].swap(bl);
+  inode_backpointer_t bp;
+  if (!op.top_op()->get_ancestor(bp))
+    return -1;
+  frag_t frag;
+  auto item = op.inodes.find(bp.dirino);
+  if (item != op.inodes.end()){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:369]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 13]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
-        }else
-            return -1;
-    }
-    return 0;
+  meta_op::sub_op* sop = op.top_op();
+  auto item = op.inodes.find(sop->ino);
+  if (item != op.inodes.end()){
+    if (_show_fn(*(item->second), op.outfile()) < 0)
+      return -1;
+  }else{
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:383]: spaces.

File: src/tools/cephfs/MetaTool.cc [hunk 14]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
-            nsop->ino_c = sop->ino;
-            op.push_op(nsop);
-            return 1;
-        }else
-            return -1;
-    }
-    return 0;
+  meta_op::sub_op* sop = op.top_op();
+  auto item = op.inodes.find(sop->ino);
+  if (item != op.inodes.end()){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:380]: space

File: src/tools/cephfs/MetaTool.cc [hunk 15]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+    if (_show_fn(*(item->second), op.outfile()) < 0)
+      return -1;
+  }else{
+    if (op.inodes.empty()){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:384]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 16]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+      meta_op::sub_op* nsop = new meta_op::sub_op(&op);
+      nsop->sub_op_t = meta_op::OP_LIST;
+      nsop->sub_ino_t = meta_op::INO_DIR;
+      nsop->trace_level = 0;
+      nsop->ino_c = sop->ino;
+      op.push_op(nsop);
+      return 1;
+    }else
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:392]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 17]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+  std::string format = "json";
+  std::string oids;
+  Formatter* f = Formatter::create(format);
+  f->enable_line_break();
+  f->open_object_section("fnodes");
+  for (const auto &frag : frags){
+    bufferlist hbl;
+    string oid = obj_name(inode_meta.get_meta()->inode.ino, frag);
+    int ret = io_meta.omap_get_header(oid, &hbl);
+    if (ret < 0){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:411]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 18]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+      oids += oid;
+      f->open_object_section(oid.c_str());
+      got_fnode.dump(f);
+      f->close_section();
+    }
+  }
+  f->dump_string("oids", oids.c_str());
+  f->close_section();
+  f->flush(ds);
+  if (fn != ""){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:436]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 19]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+    ofstream o;
+    o.open(fn);
+    if (o){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:439]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 20]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+      o << ds.str();
+      o.close();
+    }
+    else{
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:443]: please move this up, so it looks like
```c++
} else {
```

File: src/tools/cephfs/MetaTool.cc [hunk 21]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+      cout << "out to file (" << fn << ") failed" << std::endl;
+      cout << ds.str() << std::endl;
+    }
+  }else
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:447]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 22]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
-            if (ret < 0)
-                return ret;
-        }
-    } catch (JSONDecoder::err& e) {
-        cout << "failed to decode JSON input: " << e.what() << std::endl;
-        return -1;
-    }
-    return 0;
+  JSONParser parser;
+  if(!parser.parse(fn.c_str())) {
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:458]: space

File: src/tools/cephfs/MetaTool.cc [hunk 23]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+    }
+  }else
+    std::cout << ds.str() << std::endl;
+  return 0;
 }
 int MetaTool::amend_fn(meta_op &op){
-    if(_amend_fn(op.infile(), op.confirm_chg()) < 0)
-        return -1;
-    return 0;
+  if(_amend_fn(op.infile(), op.confirm_chg()) < 0)
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:452]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 24]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
-        cout << "failed to decode JSON input: " << e.what() << std::endl;
-        return -1;
-    }
-    return 0;
+  JSONParser parser;
+  if(!parser.parse(fn.c_str())) {
+    cout << "Error parsing create user response : " << fn << std::endl;
+    return -1;
+  }
+  if (!confirm){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:462]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 25]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+  }
+  try {
+    string tmp;
+    JSONDecoder::decode_json("oids", tmp, &parser, true);
+    string::size_type pos1, pos2;
+    vector<string> v;
+    string c = ",";
+    pos2 = tmp.find(c);
+    pos1 = 0;
+    while(string::npos != pos2){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:477]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 26]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+      v.push_back(tmp.substr(pos1, pos2-pos1));
+      pos1 = pos2 + c.size();
+      pos2 = tmp.find(c, pos1);
+    }
+    if(pos1 != tmp.length())
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:482]: space.

File: src/tools/cephfs/MetaTool.cc [hunk 27]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
-    auto item = op.inodes.find(sop->ino);
-    if (item != op.inodes.end()){
-        if (_show_meta(*(item->second), op.outfile()) < 0)
-            return -1;
+  meta_op::sub_op* sop = op.top_op();
+  auto item = op.inodes.find(sop->ino);
+  if (item != op.inodes.end()){
+    if (_show_meta(*(item->second), op.outfile()) < 0)
+      return -1;
+  }else{
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:507]: spaces.

File: src/tools/cephfs/MetaTool.cc [hunk 28]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
-            nsop->sub_op_t = meta_op::OP_LIST;
-            nsop->sub_ino_t = meta_op::INO_DIR;
-            nsop->trace_level = 0;
-            nsop->ino_c = sop->ino;
-            op.push_op(nsop);
-            return 1;
-        }else{
-            return -1;
-        }
+      return -1;
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:517]: spaces in the above line.

File: src/tools/cephfs/MetaTool.cc [hunk 29]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
+  f->flush(ds);
+
+  if (fn != ""){
+    ofstream o;
+    o.open(fn);
+    if (o){
+      o << ds.str();
+      o.close();
+    }
+    else{
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:559]: move this line into the above line.

File: src/tools/cephfs/MetaTool.cc [hunk 30]
```diff
@@ -333,519 +335,518 @@ void MetaTool::inode_meta_t::encode(::ceph::bufferlist& bl, uint64_t features){
...
-        }
-        else{
-            cout << "out to file (" << fn << ") failed" << std::endl;
-            cout << ds.str() << std::endl;
-        }
-            
-    }else
-        std::cout << ds.str() << std::endl;
-    return 0;
+  }else
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:564]: space.

Comment (simon-rock) [reply src/tools/cephfs/MetaTool.cc]: @tchaikov updated it.

Comment (simon-rock) [reply src/tools/cephfs/MetaTool.cc]: @tchaikov
https://github.com/ceph/ceph/pull/32422
https://github.com/ceph/ceph/pull/31514
These two patches haven't progressed for two months. review them please.
I'll fix them as soon as possible if there's a problem.
thx.

File: src/tools/cephfs/MetaTool.h [hunk 31]
```diff
@@ -16,253 +18,256 @@ using std::stack;
 using ::ceph::bufferlist;
 class MetaTool : public MDSUtility
 {
- public:
-    class inode_meta_t{
-    public:
+public:
+  class inode_meta_t{
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.h:22]: space before `{`.

File: src/tools/cephfs/MetaTool.h [hunk 32]
```diff
@@ -16,253 +18,256 @@ using std::stack;
...
-        
-        std::map<inodeno_t, inode_backpointer_t > ancestors;
-        std::map<inodeno_t, inode_meta_t* > inodes;
-        std::map<inodeno_t, string > okeys;
+        _out(out),
+        _in(in),
+        _confirm(confirm)
+      {}
+    void release();
+    typedef enum{
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.h:56]: space before `{`.

File: src/tools/cephfs/MetaTool.h [hunk 33]
```diff
@@ -16,253 +18,256 @@ using std::stack;
...
+      OP_LIST = 0,
+      OP_LTRACE,
+      OP_SHOW,
+      OP_AMEND,
+      OP_SHOW_FN,
+      OP_AMEND_FN,
+      OP_NO
+    }op_type;
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.h:64]: space after `}`.

File: src/tools/cephfs/MetaTool.h [hunk 34]
```diff
@@ -16,253 +18,256 @@ using std::stack;
...
-        bool confirm_chg(){
-            return _confirm;
-        }
-    private:
-        stack<sub_op*> sub_ops;
-        bool _debug;
-        string _out;
-        string _in;
-        bool _confirm;
+    typedef enum{
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.h:66]: ditto.

File: src/tools/cephfs/MetaTool.h [hunk 35]
```diff
@@ -16,253 +18,256 @@ using std::stack;
...
+      INO_DIR = 0,
+      INO_F
+    }ino_type;
+
+    static string op_type_name(op_type& t){
+      string name;
+      switch (t){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.h:73]: space before `{`.

File: src/tools/cephfs/MetaTool.h [hunk 36]
```diff
@@ -16,253 +18,256 @@ using std::stack;
...
-        bool _debug;
-        string _out;
-        string _in;
-        bool _confirm;
+    typedef enum{
+      INO_DIR = 0,
+      INO_F
+    }ino_type;
+
+    static string op_type_name(op_type& t){
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.h:71]: space before `{`.

Comment (simon-rock) [reply src/tools/cephfs/MetaTool.h]: updated it. thx

File: src/tools/cephfs/MetaTool.cc [hunk 37]
```diff
@@ -876,95 +903,95 @@ int MetaTool::show_child(std::string_view key,
...
-    if (_debug)
-        std::cout << pos << " type '" << type << "' dname '" << dname
-                  << " [" << first << "," << last << "]"
-                  << std::endl;
-    // bool stale = false;
-    if (snaps && last != CEPH_NOSNAP){
-        derr << "!!!! erro !!!!" << dendl;
-        return -1;
+    std::stringstream ds;
+    //std::string format = "json-pretty";
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:954]: could you take this opportunity to drop this line?

File: src/tools/cephfs/MetaTool.cc [hunk 38]
```diff
@@ -876,95 +903,95 @@ int MetaTool::show_child(std::string_view key,
...
+    std::string format = "json";
+    // Formatter f = new JSONFormatter(false);
```
Comment (tchaikov) [inline src/tools/cephfs/MetaTool.cc:956]: ditto.

File: src/tools/cephfs/cephfs-meta-injection.cc [hunk 39]
```diff
@@ -15,87 +17,88 @@ using std::string;
...
+  po::options_description all("all options");
+  all.add(modeoptions).add(general);
+  po::variables_map vm;
+  try {
+    po::store(po::command_line_parser(argc, argv).options(all).positional(p).allow_unregistered().run(), vm);
+  } catch(exception &e) {
+    cerr << "error : " << e.what() << std::endl;
+    return -1;
+  }
+  catch(...) {
```
Comment (tchaikov) [inline src/tools/cephfs/cephfs-meta-injection.cc:66]: could you move this line up into the previous line?

Comment (simon-rock) [reply src/tools/cephfs/cephfs-meta-injection.cc]: @tchaikov  updated it. thx

**Comment by ukernel** [comment]
LGTM

**Pull request merged**